lib/ecs_deploy/auto_scaler/instance_fluctuation_manager.rb

```diff
@@ -0,0 +1,165 @@
+require "aws-sdk-autoscaling"
+require "aws-sdk-ec2"
+require "aws-sdk-ecs"
+
+module EcsDeploy
+  module AutoScaler
+    class InstanceFluctuationManager
+      attr_reader :logger
+
+      def initialize(region:, cluster:, cluster_to_asg:, desired_capacity:, logger:)
+        @region = region
+        @cluster = cluster
+        @cluster_to_asg = cluster_to_asg
+        @desired_capacity = desired_capacity
+        @logger = logger
+
+        unless cluster_to_asg.key?(cluster)
+          raise ArgumentError, "Unknown cluster: #{cluster}"
+        end
+
+        @asg_name = cluster_to_asg[cluster]
+      end
+
+      def increase
+        asg = as_client.describe_auto_scaling_groups(auto_scaling_group_names: [@asg_name]).auto_scaling_groups.first
+
+        @logger.info("Increase desired capacity of #{@asg_name}: #{asg.desired_capacity} => #{asg.max_size}")
+        as_client.update_auto_scaling_group(auto_scaling_group_name: @asg_name, desired_capacity: asg.max_size)
+
+        # Run in background because increasing instances may take time
+        Thread.new do
+          loop do
+            cluster = ecs_client.describe_clusters(clusters: [@cluster]).clusters.first
+            instance_count = cluster.registered_container_instances_count
+            if instance_count == asg.max_size
```

Comment:
[imo] It might be good to make a condition for case of capacity changed, for example case which it no longer need to scale to the maximum.I think scaling max capacity is not necessarily required.